src/main/java/org/zalando/failsafeactuator/config/FailsafeInjectionConfiguration.java

```diff
@@ -34,9 +32,9 @@
 
   @Bean
   @Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
```

Comment:
Misleading because your registry actually implements a singleton pattern. Couldn't you delegate all of this work to spring?

Reply:
Yes, we could do this. But I would prefer to do this change in another PR.

Reply:
See #45